marconi/test/Integration.hs

```diff
@@ -0,0 +1,355 @@
+import Marconi.Indexers qualified as M
+import Marconi.Logging ()
+
+tests :: TestTree
+tests = testGroup "Integration"
+  [ testProperty "prop_script_hash_in_local_testnet_tx_match" testIndex ]
+
+{- | We test the script transaction indexer by setting up a testnet,
+   adding a script to it and then spending it, and then see if the
+   script shows up on the indexer.
+
+   Specifically, we:
+
+    - spin up a testnet and the script transaction indexer
+
+    - create a plutus script, then submit a transaction which contains
+      an UTxO where this script is the validator
+
+    - submit a second transaction that spends this UTxO, then query
+      the indexer to see if it was indexed properly
+-}
+testIndex :: Property
+testIndex = H.integration . HE.runFinallies . HE.workspace "chairman" $ \tempAbsBasePath' -> do
+
+  base <- HE.noteM $ liftIO . IO.canonicalizePath =<< HE.getProjectBase
+  (socketPathAbs, networkId, tempAbsPath) <- startTestnet base tempAbsBasePath'
+
+  -- Create a channel that is passed into the indexer, such that it
+  -- can write index updates to it and we can await for them (also
+  -- making us not need threadDelay)
+  indexedTxs <- liftIO IO.newChan
+  let writeScriptUpdate (ScriptTx.ScriptTxUpdate txScripts _slotNo) = case txScripts of
+        (x : xs) -> IO.writeChan indexedTxs $ x :| xs
+        _        -> pure ()
+
+  -- Start indexer
+  let sqliteDb = tempAbsPath </> "script-tx.db"
+  indexer <- liftIO $ do
+
+    coordinator <- M.initialCoordinator 1
+    ch <- IO.atomically . IO.dupTChan $ M._channel coordinator
+    (loop, indexer) <- M.scriptTxWorker_ (\_ update -> writeScriptUpdate update $> []) (ScriptTx.Depth 0) coordinator ch sqliteDb
+
+    -- Receive ChainSyncEvents and pass them on to indexer's channel
+    void $ IO.forkIO $ do
+      let chainPoint = C.ChainPointAtGenesis :: C.ChainPoint
+      c <- defaultConfigStdout
+      withTrace c "marconi" $ \trace -> let
+        indexerWorker = withChainSyncEventStream socketPathAbs networkId chainPoint $ S.mapM_ $
+          \chainSyncEvent -> IO.atomically $ IO.writeTChan ch chainSyncEvent
+        handleException NoIntersectionFound = logError trace $ renderStrict $ layoutPretty defaultLayoutOptions $
+          "No intersection found for chain point" <+> pretty chainPoint <> "."
+        in indexerWorker `catch` handleException :: IO ()
+
+    -- Start indexer worker loop
+    void $ IO.forkIO loop
+
+    return indexer
+
+  utxoVKeyFile <- H.note $ tempAbsPath </> "shelley/utxo-keys/utxo1.vkey"
+  utxoSKeyFile <- H.note $ tempAbsPath </> "shelley/utxo-keys/utxo1.skey"
+
+  let
+    -- Create an always succeeding validator script
+    plutusScript :: C.PlutusScript C.PlutusScriptV1
+    plutusScript = C.PlutusScriptSerialised $ SBS.toShort . LBS.toStrict $ serialise $ Plutus.unValidatorScript validator
+      where
+        validator :: Plutus.Validator
+        validator = Plutus.mkValidatorScript $$(PlutusTx.compile [|| \_ _ _ -> () ||])
```

Comment:
This exists as `Plutus.Script.Utils.V1.Generators.alwaysSucceedValidator`

Reply:
Thanks! ca2c08203e4e2569a565e11abe4bc546632f114b